x-pack/plugins/ml/public/application/components/anomalies_table/links_menu.tsx

```diff
@@ -298,15 +359,75 @@ export const LinksMenuUI = (props: LinksMenuProps) => {
           dataViewId === null
             ? []
             : getFiltersForDSLQuery(job.datafeed_config.query, dataViewId, job.job_id),
-      });
+        ...(withWindowParameters
+          ? {
+              wp: { bMin, bMax, dMin, dMax },
+            }
+          : {}),
+      };
+    };
+
+    const generateDiscoverUrl = async () => {
+      const pageState = await generateRedirectUrlPageState();
+      const url = await discoverLocator.getRedirectUrl(pageState);
 
       if (!unmounted) {
         setOpenInDiscoverUrl(url);
       }
     };
 
+    const generateLogRateAnalysisUrl = async () => {
+      if (
+        !LOG_RATE_ANALYSIS_ML_FUNCTIONS.includes(
+          props.anomaly.source.function as LogRateAnalysisMlFunctions
+        ) &&
+        !unmounted
+      ) {
+        setOpenInLogRateAnalysisUrl(undefined);
+        return;
+      }
+
+      const mlLocator = share.url.locators.get(ML_APP_LOCATOR);
+
+      if (!mlLocator) {
```

Comment:
Nit, in a few places we just `!` assume that `ML_APP_LOCATOR` exists. It's safe to do so as we're in the ML app. It'll only be undefined for plugins which have been disabled and so haven't registered their locator.